### PR TITLE
Add environment variable GEOPM_CTL_LOCAL

### LINKED ISSUE #3023: As a user of the GEOPM runtime I would like to run the geopmctl(1) command and opt out of using MPI
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I have compiled the GEOPM Runtime with MPI enabled, but I do not want to run the GEOPM Controller using MPI.  This may be because I either do not wish for the GEOPM runtime to utilize the MPI fabric, nor do I wish to launch the GEOPM Controller using an MPI launcher.  This would enable the exact same use case as when the GEOPM Runtime is configured with `--disable-mpi` but allowing a single binary to serve both use cases.

**Describe the solution you'd like**
Add an environment variable like `GEOPM_CTL_LOCAL` which would be used to skip calls to MPI by the `Controller`, and utilize the `NullComm` rather  than the `MPIComm` class.

**Describe alternatives you've considered**
Currently this feature is only available as a compile time option, which restricts usablity.

### PR DESCRIPTION
**Pull request by cmcantalupo**
- When set disable controller use of MPI
- Fixes #3023

### CODE REVIEW

File: src/Controller.cpp [hunk 1]
```diff
@@ -63,12 +66,32 @@ extern "C"
...
+                    char error_str[NAME_MAX + 1] = {};
+                    PMPI_Error_string(err, error_str, &str_size);
+                    fprintf(stderr, "Error: %s\n", error_str);
+                    return err;
+                }
+            }
+#endif
+            {
+                std::unique_ptr<geopm::Controller> ctl = std::make_unique<geopm::Controller>();
+                err = geopm_ctl_run((struct geopm_ctl_c *)ctl.get());
```
Comment (cmcantalupo) [inline src/Controller.cpp]: maybe should just be calling  ``ctl->run()`` no need for the C interface anymore.

File: src/Controller.cpp [hunk 2]
```diff
@@ -63,12 +66,32 @@ extern "C"
...
         int err = 0;
+        bool do_ctl_local = geopm::environment().do_ctl_local();
         try {
-            geopm::Controller ctl;
-            err = geopm_ctl_run((struct geopm_ctl_c *)&ctl);
+#ifdef GEOPM_ENABLE_MPI
+            if (!do_ctl_local) {
+                err = PMPI_Init(&argc, &argv);
+                if (err) {
+                    int str_size = NAME_MAX;
```
Comment (dannosliwcd) [inline src/Controller.cpp]: Should this be `MPI_MAX_ERROR_STRING` instead?

File: src/Environment.cpp [hunk 3]
```diff
@@ -214,7 +215,11 @@ namespace geopm
 
     std::string EnvironmentImp::comm(void) const
     {
-        return lookup("GEOPM_COMM");
+        std::string ret = "NullComm";
+        if (!is_set("GEOPM_CTL_LOCAL")) {
```
Comment (avilcheslopez) [inline src/Environment.cpp]: Suggestion: leverage `do_ctl_local()`

File: service/docs/source/geopmlaunch.1.rst [hunk 4]
```diff
@@ -553,6 +553,13 @@ GEOPM Options
                      launch the GEOPM runtime in process mode.  When this
                      option is specified, the GEOPM runtime will not be
                      launched.
+--geopm-ctl-local  .. _geopm-ctl-local option:
+
+                   Disable all communication between controllers
+                   running on different compute nodes.  This will
+                   result in one report file per host, and each report
+                   file will have the ``hostname`` appended to the file
+                   path requested by the user.
```
Comment (cmcantalupo) [inline service/docs/source/geopmlaunch.1.rst]: Add comment that this option has no effect when --geopm-ctl is set to process or pthread.

File: src/Controller.cpp [hunk 5]
```diff
@@ -63,12 +66,33 @@ extern "C"
...
+                    PMPI_Error_string(err, error_str, &str_size);
+                    fprintf(stderr, "Error: %s\n", error_str);
+                    return err;
+                }
+            }
+#endif
+            {
+                std::unique_ptr<geopm::Controller> ctl = std::make_unique<geopm::Controller>();
+                ctl->run();
+            }
```
Comment (bgeltz) [inline src/Controller.cpp:90]: Was the extra {} here necessary for something?

**Review by bgeltz — APPROVED** [review]

File: src/Controller.cpp (hunk 5 above)
Comment (cmcantalupo) [reply src/Controller.cpp:90]: that forces the controler destructor to be called, which needs to happen before MPI_FInalize()

**Pull request merged**